judge/views/problem_data.py

```diff
@@ -178,26 +179,35 @@ def get_valid_files(self, data, post=False):
             elif data.zipfile:
                 return ZipFile(data.zipfile.path).namelist()
         except BadZipfile:
-            return None
+            raise
```

Comment:
Why are you catching this one exception only and reraising it? That's equivalent to not catching it since you don't have any subsequent `except` clauses.

Reply:
I know, my point was to be explicit about letting the exception escape the function.